### PR TITLE
Initial Project Setup with Firebase Authentication for Login and Signup

### PR DESCRIPTION
**Pull request by Domm6**
This pull request establishes the foundational structure of the project and integrates Firebase Authentication. It includes the creation of the initial pages for user login and signup. Users can now register and log in using their email and password.

### CODE REVIEW

**Comment by wspalding** [comment]
Did you intend to commit all of the node modules?

**Comment by jzwang43** [comment]
let's add gitignore to exclude node_modules: https://github.com/toptal/gitignore/blob/master/templates/react.gitignore

File: Frontend/src/App.css
Comment (wspalding) [inline Frontend/src/App.css:1]: ideally we should keep the css changes in separate commits from the gitIgnore changes, but not a big deal

**Comment by jzwang43** [comment]
Is there any test plan you can share to show what's working? But overall looks good. If there's nothing renderable it's fine.

**Review by jzwang43 — APPROVED** [review]
LGTM

File: Frontend/src/index.css [hunk 1]
```diff
@@ -1,4 +1,4 @@
 body {
-  margin: 0px;
+  margin: -10px;
```
Comment (sbrina813) [inline Frontend/src/index.css:2]: margin: -10px feels a little odd. Is margin:0 with padding: 10px works for your styling?

File: Frontend/src/App.css
Comment (Domm6) [reply Frontend/src/App.css:1]: Will keep in mind when committing in the future.

File: Frontend/src/index.css (hunk 1 above)
Comment (Domm6) [reply Frontend/src/index.css:2]: I used this styling in order to fix the extra margin which caused a border around all my elements for some reason. When margin is at 0px, there is still a border.

**Comment by Domm6** [comment]
Below I have two screenshots of what it looks like as of right now. Still working on styling and details:

Screenshot 2024-06-25 at 11 14 39 AM
Screenshot 2024-06-25 at 11 14 55 AM
